modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb

```diff
@@ -98,6 +102,22 @@ def fetch_all_cached_or_service(validated_params, show_completed)
         [Mobile::V0::ClaimOverviewSerializer.new(list, options), status]
       end
 
+      def get_accessible_claims_appeals
+        if claims_access
```

Comment:
Actually, looking at this again, it looks like it looks like we're checking lighthouse access policy twice in index.

Reply:
I think we can remove the index only before_action totally since both policies are already being checked

Reply:
Nice catch, `claims_access` does appear to be redundant with the `before_action` check. I think it's better to leave the before action though and remove the `claims_access` check in this function. It's much easier to read what authorization is in place for an endpoint if it's in the before action, rather than needing to follow the call stack to find what's being checked

Reply:
yeah that makes sense. can remove the appeals access from the before_action and the claims_access if statement in the method itself.

Reply:
Should be checked by authed services too so we should be save to do that

Reply:
Wouldn't the `appeals_access` still need to be in this function though because the function return two completely different results depending on the `appeal_access` response?

Reply:
In the before action block for index, it returns either the name of service (if user has access) or 403 error.
In the claim_access & appeal_access methods it returns either true or false to then use to determine whether to pull claims, appeals, or both.
 Do you guys know how that can be simplified so that I wouldn't need the extra check in the claim_access & appeal_access methods?

Reply:
I'm thinking the best way to create some instance variables called `claims_access` and `appeals_access` that get set in the before action. that way, we don't have to worry about errors getting thrown when running `get_accessible_claims_appeals` method

Reply:
There's a lot going on here, so first let's all get on the same page about how these auth methods work, because I think it's non-obvious. 

When we call `authorize` in the controller, we're really calling `Pundit::Authorization#authorize`, which is being inherited from `SignIn::ApplicationController`. That method ultimately does this:
```
      raise NotAuthorizedError, query: query, record: record, policy: policy unless policy.public_send(query)

      record
```
So it either raises an error or returns the policy name.

When we call `user.authorize`, we're really calling the `Authorization#authorize` (i.e., from our `Authorization` module), which calls `Pundit::Authorization#policy!`. That method does this:
```
    def policy!(user, record)
      policy = PolicyFinder.new(record).policy!
      policy.new(user, pundit_model(record))
    rescue ArgumentError
      raise InvalidConstructorError, "Invalid #<#{policy}> constructor is called"
    end
```
As Chika mentioned, that will return true or false without raising an auth error. 

So calling both of these methods isn't exactly redundant because one of them is being used in conjunction with our error handling module to return 403 while the other is used to determine which upstreams we pull data from. 

In the before action above:
```
      before_action(only: %i[index]) do
        lighthouse? ? authorize(:lighthouse, :access?) : authorize(:evss, :access?)
      rescue Pundit::NotAuthorizedError
        authorize :appeals, :access?
      end
```
we're using the rescue effectively to say, "if the user doesn't have access to claims, check to see if they have access to appeals". It's technically accurate because the user only needs access to one of the resources, but using rescue isn't ideal. I think we can refactor this to simplify the code. First, get rid of the before_action then we can abstract the service into a method:

```
def service
  Flipper.enabled?(:mobile_lighthouse_claims, @current_user) ? lighthouse_claims_proxy : evss_claims_proxy
end
```

Then this method becomes:
```
if claims_access && appeals_access
  # get both
elsif claims_access
  ...
elsif appeals_access
  ***
else
  raise NotAuthorizedError # may need to add additional details to the error
end
```

Reply:
also, can `get_accessible_claims_appeals` can look something like 

```
if Flipper.enabled?(:mobile_lighthouse_claims, @current_user)
    if claims_access and appeals_access
      return lighthouse_claims_proxy.get_claims_and_appeals
    end

    if claims_access
      return lighthouse_claims_proxy.get_claims
    end
     
    evss_claims_proxy.get_appeals

  else
    if claims_access and appeals_access
      return evss_claims_proxy.get_claims_and_appeals
    end

    if claims_access
      return evss_claims_proxy.get_claims
    end

    evss_claims_proxy.get_appeals
end 
```

could put the redundant if statements into it's own method and pass it the right proxy but we'll be ripping out the evss code soon enough so I don't mind either way.